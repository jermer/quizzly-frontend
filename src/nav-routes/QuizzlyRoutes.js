import { useContext } from "react";
import { Routes, Route } from "react-router-dom";

import Homepage from "../homepage/Homepage";
import LoginForm from "../auth/LoginForm";
import SignupForm from "../auth/SignupForm";
import ProfileForm from "../profile/ProfileForm";

// import QuizList from "../quizzes/QuizList";
import LibraryList from "../library/LibraryList";
import QuizEditor from "../editor/QuizEditor";

import DiscoverList from "../player/DiscoverList";
import Player from "../player/Player";

import ReportList from "../reports/ReportList";

const QuizzlyRoutes = ({ login, signup }) => {

    return (
        <Routes>
            <Route path="/" element={<Homepage />} />
            <Route path="login" element={<LoginForm login={login} />} />
            <Route path="signup" element={<SignupForm signup={signup} />} />

            {/* <Route path="/discover" element={<QuizList />} /> */}
            <Route path="/discover" element={<DiscoverList />} />
            <Route path="/player/:id" element={<Player />} />

            <Route path="/creator/library" element={<LibraryList />} />
            <Route path="/creator/:id" element={<QuizEditor />} />

            <Route path="/reports" element={<ReportList />} />

            <Route path="/profile" element={<ProfileForm />} />
        </Routes>
    );

}

export default QuizzlyRoutes;