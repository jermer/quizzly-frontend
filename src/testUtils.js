import UserContext from "./auth/UserContext";

const demoUser = {
    username: "testuser",
    email: "test@test.net",
};

// what about "setCurrentUser"

const UserProvider =
    ({ children, currentUser = demoUser }) => (
        <UserContext.Provider value={{ currentUser }}>
            {children}
        </UserContext.Provider>
    );

export { UserProvider };
