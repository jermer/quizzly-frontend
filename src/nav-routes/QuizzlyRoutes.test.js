import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { UserProvider } from '../testUtils';
import QuizzlyRoutes from "./QuizzlyRoutes";

// smoke test
it('renders without crashing', () => {
    render(
        <MemoryRouter>
            <UserProvider>
                <QuizzlyRoutes />
            </UserProvider>
        </MemoryRouter>
    );
});

// snapshot tests
it("matches snapshot", function () {
    const { asFragment } = render(
        <MemoryRouter>
            <UserProvider>
                <QuizzlyRoutes />
            </UserProvider>
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});