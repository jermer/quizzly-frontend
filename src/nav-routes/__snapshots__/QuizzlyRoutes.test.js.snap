// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <h1>
    QUIZZLY IS UP AND RUNNING!
  </h1>
</DocumentFragment>
`;
